src/odemis/driver/xt_client.py

```diff
@@ -106,6 +232,50 @@ def __init__(self, name, role, children, address, daemon=None,
                           "uri is correct and XT server is"
                           " connected to the network. %s" % (address, err))
 
+        # Steps to transfer latest xtadapter package if available
+        # The transferred package will be a zip file in the form of bytes
+        try:
+            # xtadapter debian package installation directory which contains xtadapter's zip files
+            package = None
+            install_dir = "/usr/share/xtadapter"
+            bitness = re.search("bitness:\s*([\da-z]+)", self._swVersion)
+            bitness = bitness.group(1) if bitness is not None else None
+            adapter = "xtadapter"
+            if "xttoolkit" in self._swVersion:
+                adapter = "fastem-xtadapter"
+            current_version = re.search("xtadapter:\s*([\d.]+)", self._swVersion)
+            current_version = current_version.group(1) if current_version is not None else None
+            if current_version is not None and bitness is not None:
+                package = check_latest_package(
+                    directory=install_dir,
+                    current_version=current_version,
+                    adapter=adapter,
+                    bitness=bitness,
+                    is_zip=True,
+                )
+            if package is not None:
+                zip_file = zipfile.ZipFile(package.path)
+                ret = zip_file.testzip()
+                zip_file.close()
+                if ret is None:
+                    # Open the package's zip file as bytes and transfer them
+                    with open(package.path, mode="rb") as f:
+                        data = f.read()
+                        self.transfer_latest_package(data)
+                        f.close()
```

Comment:
Why not just ending this `with` block here? The rest of the block can be done outside of the `with`. Then you don't need to explicitly close the file.